staging/src/k8s.io/legacy-cloud-providers/azure/clients/azure_client_config.go

```diff
@@ -26,6 +26,7 @@ import (
 
 // ClientConfig contains all essential information to create an Azure client.
 type ClientConfig struct {
+	CloudName               string
```

Comment:
why would you add a `CloudName` field here? is it necessary?

Reply:
It's necessary because without doing it, I don't see an easy way to get the cloud name inside the client constructor.

Reply:
could you squash all these 3 commits by `git rebase -i HEAD~3`

Reply:
Done. Thanks!